api/src/org/apache/cloudstack/api/response/ZoneResponse.java

```diff
@@ -233,6 +234,9 @@ public void addTag(ResourceTagResponse tag) {
     }
 
     public void setResourceDetails(Map<String, String> details) {
-        this.resourceDetails = details;
+        if (details == null) {
```

Comment:
are you solving an extra bug here? seems like it is no longer possible to set details to null. I suppose it is intended.